File: services/notification-service/src/main/java/com/tap2eat/notification/messaging/events/OrderStatusChangedEvent.java
package com.tap2eat.notification.messaging.events;

import com.fasterxml.jackson.annotation.JsonAlias;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

@JsonIgnoreProperties(ignoreUnknown = true)
public record OrderStatusChangedEvent(
        @JsonProperty("EventId") @JsonAlias("eventId") String eventId,
        @JsonProperty("EventType") @JsonAlias("eventType") String eventType,
        @JsonProperty("OrderId") @JsonAlias("orderId") String orderId,
        @JsonProperty("CustomerAccountId") @JsonAlias("customerAccountId") String customerAccountId,
        @JsonProperty("RestaurantId") @JsonAlias("restaurantId") String restaurantId,
        @JsonProperty("BranchId") @JsonAlias("branchId") String branchId,
        @JsonProperty("PreviousStatus") @JsonAlias("previousStatus") String previousStatus,
        @JsonProperty("NewStatus") @JsonAlias({"newStatus", "status"}) String newStatus,
        @JsonProperty("EstimatedPreparationMinutes") @JsonAlias("estimatedPreparationMinutes") Integer estimatedPreparationMinutes,
        @JsonProperty("EstimatedReadyAt") @JsonAlias("estimatedReadyAt") String estimatedReadyAt,
        @JsonProperty("OccurredAt") @JsonAlias("occurredAt") String occurredAt
) {
}

File: services/order-service/Messaging/Events/OrderStatusChangedEvent.cs
namespace OrderService.Messaging.Events;

public sealed class OrderStatusChangedEvent
{
    public Guid EventId { get; set; } = Guid.NewGuid();

    public string EventType { get; set; } = "order.status.changed";

    public string OrderId { get; set; } = string.Empty;

    public string CustomerAccountId { get; set; } = string.Empty;

    public string RestaurantId { get; set; } = string.Empty;

    public string BranchId { get; set; } = string.Empty;

    public string PreviousStatus { get; set; } = string.Empty;

    public string NewStatus { get; set; } = string.Empty;

    public int? EstimatedPreparationMinutes { get; set; }

    public DateTime? EstimatedReadyAt { get; set; }

    public List<OrderStatusChangedItemEvent> Items { get; set; } = [];

    public DateTime OccurredAt { get; set; } = DateTime.UtcNow;
}

public sealed class OrderStatusChangedItemEvent
{
    public string ProductId { get; set; } = string.Empty;

    public int Quantity { get; set; }

    public string? ProductNameSnapshot { get; set; }
}
